org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java

```diff
@@ -11,92 +11,177 @@
+import org.eclipse.scanning.api.points.Scalar;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
 
 /**
+ * A class to generate take a set of scannables which represent simple slits then write to a nexus file
+ * as the positions are set during the scan.
  * 
- * See http://confluence.diamond.ac.uk/pages/viewpage.action?pageId=37814632
- * 
- <pre>
- primary_slit:NXslit
-    x_gap = {NX_NUMBER}
-        @local_name = "s1gapX"
-        @units = "mm"
-        @controller_record = {NX_CHAR} //EPICS name
-    y_gap = {NX_NUMBER}
-        @local_name = "s1gapY"
-        @units = "mm"
-        @controller_record = {NX_CHAR} //EPICS name
-    transforms:NXtransformations
-        x_centre = {NX_NUMBER}
-            @transformation_type = "translation"
-            @vector = 1,0,0,
-            @depends_on = "y_centre"
-            @units = "mm"
-            @controller_record = {NX_CHAR} //EPICS name
-        y_centre = {NX_NUMBER}
-            @transformation_type = "translation"
-            @vector = 0,1,0,
-            @depends_on = "."
-            @offset = 0,0,-14500 //This may change
-            @units = "mm"
-            @controller_record = {NX_CHAR} //EPICS name
-    beam:NXbeam
-        //Removed distance from here, since it's in the "y_centre" above
-        incident_beam_divergence[2,i] = {NX_FLOAT}
-            @units = "radians"
-        final_beam_divergence[2,i] = {NX_FLOAT}
-            @units = "radians"
-    motors:NXcollection
-        downstream_x:NXpositioner
-            name = "s1dsX"
-            description = "Downstream X position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        downstream_y:NXpositioner
-            name = "s1dsY"
-            description = "Downstream Y position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        upstream_x:NXpositioner
-            name = "s1usX"
-            description = "Upstream X position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        upstream_y:NXpositioner
-            name = "s1usY"
-            description = "Upstream Y position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
- </pre>
- 
- * 
- * @author Matthew Gerring
- *
+ * @see {@link MockScannableConfiguration}
  */
-public class MockNeXusSlit extends MockNeXusScannable {
+public class MockNeXusSlit extends MockScannable implements INexusDevice<NXslit> {
+
+	public boolean isWritingOn() {
+		return writingOn;
+	}
+
+	public void setWritingOn(boolean writingOn) {
+		this.writingOn = writingOn;
+	}
+
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)